Sources/StreamChat/ChatClient.swift

```diff
@@ -516,24 +446,197 @@ extension ChatClient: ConnectionDetailsProviderDelegate {
     }
 }
 
+extension ClientError {
+    public class MissingLocalStorageURL: ClientError {
+        override public var localizedDescription: String { "The URL provided in ChatClientConfig is `nil`." }
+    }
+
+    public class ConnectionNotSuccessful: ClientError {
+        override public var localizedDescription: String {
+            """
+            Connection to the API has failed.
+            You can read more about making a successful connection in our docs:
+            https://getstream.io/chat/docs/sdk/ios/uikit/getting-started/
+            \n
+            API Error: \(String(describing: errorDescription))
+            """
+        }
+    }
+
+    public class MissingToken: ClientError {}
+    class WaiterTimeout: ClientError {}
+
+    public class ClientIsNotInActiveMode: ClientError {
+        override public var localizedDescription: String {
+            """
+                ChatClient is in connectionless mode, it cannot connect to websocket.
+                Please check `ChatClientConfig.isClientInActiveMode` for additional info.
+            """
+        }
+    }
+
+    public class ConnectionWasNotInitiated: ClientError {
+        override public var localizedDescription: String {
+            """
+                Before performing any other actions on chat client it's required to connect by using \
+                one of the available `connect` methods e.g. `connectUser`.
+            """
+        }
+    }
+
+    public class ClientHasBeenDeallocated: ClientError {
+        override public var localizedDescription: String {
+            "ChatClient has been deallocated, make sure to keep at least one strong reference to it."
+        }
+    }
+
+    public class MissingTokenProvider: ClientError {
+        override public var localizedDescription: String {
+            """
+                Missing token refresh provider to get a new token
+                When using expiring tokens you need to provide a way to refresh it by passing `tokenProvider` when \
+                calling `ChatClient.connectUser()`.
+            """
+        }
+    }
+}
+
+class ChatClientFactory {
```

Comment:
This is nice 👍 Maybe move it in a separate file?

Reply:
Yes I can move it to a separate file 👍

Reply:
✅ I've also moved the Environemnt struct to a new file as well, to make it consistent.